tests/integration/pilot/common/routing.go

```diff
@@ -155,13 +157,11 @@ spec:
 			opts: echo.CallOptions{
 				PortName: "http",
 				Count:    1,
-				Validator: echo.And(
-					echo.ExpectOK(),
-					echo.ValidatorFunc(
-						func(response echoclient.ParsedResponses, _ error) error {
-							return response.Check(func(_ int, response *echoclient.ParsedResponse) error {
-								return ExpectString(response.RawResponse["Istio-Custom-Header"], "user-defined-value", "request header")
-							})
+				Checker: check.And(
```

Comment:
super nit: its an extremely common pattern to have `And(ok, some-checker`). We could introduce a struct field that always does this (eg `CheckerOk: some-checker`), a helper (`Checker: check.OKAnd(some-checker)`), or even make checkall smarter and somehow always do "OK" if there is not an explicit "expect error"?

Reply:
I think I'll defer this particular change for now. We can always add this later.